### ISSUE TITLE
[v2.0.0-beta3][CONNECTION] App connection issue to the RC server

### ISSUE THREAD

**Issue by rafaelks** [RocketChat/Rocket.Chat#10363]
@mmarif4u commented on [Wed Apr 04 2018](https://github.com/RocketChat/Rocket.Chat.Android/issues/1019)

Your Rocket.Chat.Android version: 2.0.0 beta 3

Last time i bring this issue to the RC open chat android channel too. And that issue continues, where the app trying to connect to the RC server. While connecting it crash the RC server instance too. I have checked the server logs and found this at the time when it was attempting to connect.

```
"http://127.0.0.1:3000/avatar/@CHANNEL?format=jpeg", host: "team.WEBSITE.com"
2018/04/04 13:06:05 [error] 8297#8297: *121687 connect() failed (111: Connection refused) while connecting to upstream, client: xxx.yy.xxx.yy, server: team.WEBSITE.com, request: "GET /avatar/@CHANNEL?format=jpeg HTTP/1.1", upstream: "http://127.0.0.1:3000/avatar/@CHANNEL?format=jpeg", host: "team.WEBSITE.com"
2018/04/04 13:06:05 [error] 8297#8297: *121687 connect() failed (111: Connection refused) while connecting to upstream, client: xxx.yy.xxx.yy, server: team.WEBSITE.com, request: "GET /avatar/@CHANNEL?format=jpeg HTTP/1.1", upstream: "http://127.0.0.1:3000/avatar/@CHANNEL?format=jpeg", host: "team.WEBSITE.com"
```

Something to do with retrieving channel/user avatars?. The app also some time shows JSON errors while connecting. Which i assume is related to the json output from the API call.

Android logs, https://open.rocket.chat/channel/androidnativeapp?msg=ACg2JZC6EdKpKcgYi

@rafaelks

---

@mmarif4u commented on [Wed Apr 04 2018](https://github.com/RocketChat/Rocket.Chat.Android/issues/1019#issuecomment-378805705)

RC snap is updated to 0.63.0, and after checking this morning, this is the error now:

`
2018/04/05 10:48:02 [error] 8297#8297: *135538 connect() failed (111: Connection refused) while connecting to upstream, client: xxx.yyy.xx.yy, server: team.WEBSITE.com, request: "GET /sockjs/info?cb=izp3r5sdg2eh HTTP/1.1", upstream: "http://127.0.0.1:3000/sockjs/info?cb=izp3r5sdg2eh", host: "team.WEBSITE.com", referrer: "https://team.WEBSITE.com/channel/CHANNEL-NAME"
`

---

@HammyHavoc commented on [Thu Apr 05 2018](https://github.com/RocketChat/Rocket.Chat.Android/issues/1019#issuecomment-378913113)

I get endless "connecting" with countdown messages on my Android phone before it eventually crashes.

---

@ccfiel commented on [Thu Apr 05 2018](https://github.com/RocketChat/Rocket.Chat.Android/issues/1019#issuecomment-379122072)

same here :(

---

@rafaelks commented on [Fri Apr 06 2018](https://github.com/RocketChat/Rocket.Chat.Android/issues/1019#issuecomment-379220602)

@mmarif4u @HammyHavoc @ccfiel Thank you guys for opening the issue! So the problem is that your instance is dying when connecting to the Android app? I'm imagining that the issue related to the request that we're doing for the image in *JPEG* format. If that's the case, I'll move this issue to our back-end repository. Can you please confirm?

---

@HammyHavoc commented on [Fri Apr 06 2018](https://github.com/RocketChat/Rocket.Chat.Android/issues/1019#issuecomment-379221024)

@rafaelks Spot on! :- ) This is correct.

**Comment by ccfiel** [comment]
@rodrigok I can give you access. Where can I send the credentials?

**Comment by mmarif4u** [comment]
@rodrigok I have provided the logs in the thread above. That are the exact logs when this happens. Hope it is of some help to debug.

**Comment by ccfiel** [comment]
@rodrigok I have email you the credentials. Hope it can help

**Comment by rodrigok** [comment]
@ccfiel please contact me at https://open.rocket.chat/direct/rodrigo.nascimento

**Comment by rafaelks** [comment]
@rodrigok @sampaiodiego Added to 0.64.0 but I would say we need a patch in order to fix this one.

**Comment by geekgonecrazy** [comment]
Fixed in: #10390

Triggered a build will do quick test and push it out before 0.64.0

**Comment by geekgonecrazy** [comment]
Please `sudo snap refresh rocketchat-server` it should now be fixed

**Comment by donzavitz** [comment]
this fixed my issue #1057 but is it normal i can't send video or audio messages in this android client?

**Comment by ccfiel** [comment]
fix :)
